src/main.c

```diff
@@ -453,12 +453,73 @@ static int dev_opt(const char *str) {
     !!strchr(str, '/');
 }
 
+static int cmp_ptr(const void *a, const void *b) {
+   return (*(int*)a - *(int*)b);
+}
+
+static int suggest_programmers(const char *programmer, LISTID programmers) {
+  const int max_distance = 3;
+
+  int idx = 0;
+  int pgmid_maxlen = 0;
+  typedef struct {
+    int distance;
+    char *pgmid;
```

Comment:
I suspect if pgmid and desc are `const char *` these can be directly assigned below (no need for `cfg_strdup()` and no need for a later `free()`).

Reply:
Thanks! Fixed